src/tests/news/personalized-mock.test.ts

```diff
@@ -0,0 +1,91 @@
+import type { Request, Response } from 'express';
+import { fetchFromNewsAPI, formatArticlesResponse } from '../../utils.js';
+import newsController from '../../modules/News/controller.js';
+
+jest.mock('../../utils.js');
+
+const { getPersonalizedNews } = newsController;
+
+describe('getPersonalizedNews', () => {
+  let req: Request;
+  let res: Response;
+  const apiMock = fetchFromNewsAPI as jest.Mock;
+  const formatMock = formatArticlesResponse as jest.Mock;
+
+  beforeEach(() => {
+   
+    jest.clearAllMocks();
+
+    req = {
+      query: {},
+    } as unknown as Request;
+
+    res = {
+      status: jest.fn().mockReturnThis(),
+      send: jest.fn(),
+    } as unknown as Response;
+  });
+
+  it('should respond with 500 if api response with error', async () => {
+    apiMock.mockRejectedValue('reject-with-error')
```

Comment:
When throwing `mockRejectedValue`, it is better to use the typical throwing an error to align with the typical error handling like: `apiMock.mockRejectedValue(new Error('Error in API response'))`
[mockFn.mockRejectedValue(value)](https://jestjs.io/docs/mock-function-api#mockfnmockrejectedvaluevalue)